clouddriver-security/src/test/groovy/com/netflix/spinnaker/clouddriver/security/DefaultAllowedAccountsValidatorSpec.groovy

```diff
@@ -66,6 +68,57 @@ class DefaultAllowedAccountsValidatorSpec extends Specification {
     0 * errors.rejectValue(_, _, _)
   }
 
+  void "should allow if all accounts intersection with required group memberships"() {
+    given:
+    def errors = Mock(Errors)
+    def validator = new DefaultAllowedAccountsValidator(Mock(AccountCredentialsProvider) {
+      1 * getAll() >> { [credentialsWithRequiredGroup] }
+      0 * _
+    })
+
+    when:
+    def description = new MultiAccountDescription(credentialsList: [credentialsWithRequiredGroup, secondCredentials])
+
+    validator.validate("TestUser", ["TestAccount", "OtherAccount"], description, errors)
+
+    then:
+    0 * errors.rejectValue(_, _, _)
+  }
+
+  void "should reject if any account does not intersect with required group memberships"() {
```

Comment:
Suspect you could combine all your new tests and parameterize with `where:`

```
where:
requiredCredentials                              | userAccounts                     || failureCount
[credentialsWithRequiredGroup, thirdCredentials] | ["TestAccount", "OtherAccount"]  || 1
[credentialsWithRequiredGroup, thirdCredentials] | ["ProdAccount", "RandomAccount"] || 2
```

Would just cut down on duplication across tests.